clang/test/Parser/cxx-invalid-function-decl.cpp

```diff
@@ -40,3 +40,13 @@ struct S : public Base1<int>, public Base2<float> {
   // All initializers are correct, nothing to skip, diagnose 2 missing commas.
   S(const S &) : Base1<int>(0) ::Base2<float>(1.0) x(2) {} // expected-error2{{missing ',' between base or member initializers}}
 };
+
+namespace GH113722 {
+struct S {
+  void m(int x = 0;   // expected-error {{unexpected end of default argument expression}} \
+                         expected-note {{to match this '('}}
+    #pragma unused(x) // expected-error {{expected ')'}} \
```

Comment:
Does this mean we can't have pragmas in the middle of a statement?  It seems that something like the above is actually useful/a shame to lose. It would be wonderful if instead the ConsumeAndStoreInitializer just 'did the pragma parsing and came back', but I presume that is a lot more work?

Reply:
> Does this mean we can't have pragmas in the middle of a statement?

Nope, this still works fine:
```
struct S {
  void m(int x =
  #pragma clang diagnostic push
  #pragma clang diagnostic ignored "-Wconversion"
  1.2f
  #pragma clang diagnostic pop
  );
  void n(int x =
  1.2f
  );
};
```
I can add a test case for that if you want. This is just an issue with an ill-formed initializer, I believe.

Reply:
OH! missed that this was a function parameter/default argument and part of one that is already ill-formed.  I thinks this recovery makes sense, I don't see any need for other tests